Assets/Plugins/Editor/Uplift/UpdateTool/Updater.cs

```diff
@@ -0,0 +1,131 @@
+using UnityEngine;
+using UnityEditor;
+using System.IO;
+using System.Collections;
+using System.Collections.Generic;
+using Uplift.Common;
+using Uplift.Windows;
+using System.Net;
+using System.Linq;
+using System;
+using System.Globalization;
+
+namespace Uplift.Updating
+{
+    public class Updater : MonoBehaviour
+    {
+        private static readonly string dateFormat = @"MM\/dd\/yyyy HH:mm";
+        private static readonly CultureInfo provider = CultureInfo.InvariantCulture;
+        private static readonly string lastUpdateCheckKey = "UpliftLastUpdateCheck";
+        private static IEnumerator updateCoroutine;
+
+        public static void UpdateUplift(string url)
+        {
+            string unitypackageName = url.Split('/').Last();
+            DirectoryInfo assetsPath = new DirectoryInfo(Application.dataPath);
+            string destination = Path.Combine(assetsPath.Parent.FullName, unitypackageName);
+            
+            ServicePointManager.ServerCertificateValidationCallback = delegate { return true; };
+            using (var client = new WebClient())
+            {
+                client.DownloadFile(url, destination);
+            }
+
+            AssetDatabase.ImportPackage(destination, false);
+        }
+
+        public static void CheckForUpdate(bool forceCheck = false)
+        {
+            if(forceCheck || ShouldCheck())
+            {
+                EditorApplication.update += EditorUpdate;
+    			updateCoroutine = CheckForUpdateRoutine();
+            }
+        }
+        
+        private static void EditorUpdate()
+        {
+            updateCoroutine.MoveNext();
+        }
+
+        private static bool ShouldCheck()
+        {
+            string lastCheck = EditorPrefs.GetString(lastUpdateCheckKey);
+            if(string.IsNullOrEmpty(lastCheck)) return true;
+            
+            try
+            {
+                TimeSpan timeFromLastCheck = DateTime.UtcNow - DateTime.ParseExact(lastCheck, dateFormat, provider);
+                return timeFromLastCheck > TimeSpan.FromDays(1);
+            }
+            catch(FormatException e)
+            {
+                Debug.LogErrorFormat("The date stored for the last check of Uplift ({0}) is not in the correct format:\n{1}", lastCheck, e);
+            }
+            return true;
+        }
+        
+        private static IEnumerator CheckForUpdateRoutine() {
+			IEnumerator e = GetReleasesJson();
+			while (e.MoveNext()) yield return e.Current;
+
+			string json = (string) e.Current;
+
+			if (json == null) {
+				Debug.LogError ("Unable to check for Uplift updates");
+			} else {
+				System.Object obj = MiniJSON.Json.Deserialize (json);
+				List<System.Object> releases = (List<System.Object>)obj;
+				foreach (Dictionary<string, object> release in releases)
+				{
+                    if(VersionParser.GreaterThan((string)release ["tag_name"], About.Version))
```

Comment:
isn't tag_name formatted like "v1.0.0beta1".

Do we need to strip the v?

Reply:
Nope! The Version parser ignore the `v` by default. I have tested this and it works.